src/core/network.service.ts

```diff
@@ -136,7 +136,9 @@ export class NetworkService implements OnApplicationBootstrap {
 
   @Cron(CronExpression.EVERY_10_MINUTES)
   async removeExpiredAgreements() {
-    await this.checkControllerReady();
+    const isReady = await this.checkControllerReady();
```

Comment:
I still want to make the first execution pass, otherwise after restart we will have to wait a full cronjob iteration.

Reply:
But if controller is not ready, all the `execution` will be failed, so what's the reason have to execution it.

Reply:
We can wait until controller is ready.